### PR TITLE
feat: introduce team plan to dashboard

### PR DESCRIPTION
**Pull request by kevcodez**
Introduces the self-serve up/downgrade for the Team tier

Depends on https://github.com/supabase/infrastructure/pull/10812

### CODE REVIEW

**Review by alaister — CHANGES_REQUESTED** [review]
This looks great! I tried it out, and it all seems to be working 🎉

There's one change to do with Tailwind that I've detailed below. I'd also like @joshenlim to take a look at this when he gets back, as he knows the most about the billing section of the dashboard.

File: studio/components/interfaces/Billing/PlanSelection/PlanSelection.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ const PlanSelection: FC<Props> = ({ visible, tiers, currentPlan, onSelectPlan })
       <div className="space-y-8">
         <h4 className="text-lg">Change your project's subscription</h4>
         {/* FE will make a call to fetch all plans first at the page level */}
-        <div className="grid lg:grid-cols-3 gap-8 py-8">
+        <div className={`grid lg:grid-cols-${tiers.length} gap-8 py-8`}>
```
[the remark this answers is not in the document]
Comment (kevcodez) [reply studio/components/interfaces/Billing/PlanSelection/PlanSelection.tsx]: Adjusted

File: studio/components/interfaces/Billing/PlanSelection/PlanSelection.utils.ts [hunk 2]
```diff
@@ -17,5 +17,8 @@ export const formatTierOptions = (tiers: any[]) => {
   return tiers
     .filter((tier: any) => tier.id === STRIPE_PRODUCT_IDS.FREE)
     .concat(tiers.filter((tier: any) => tier.id === STRIPE_PRODUCT_IDS.PRO))
+    .concat(tiers.filter((tier: any) => tier.id === STRIPE_PRODUCT_IDS.TEAM))
     .concat([enterpriseOption])
+    // Team plan might not be enabled yet, make sure to filter out null/undefined
+    .filter((it) => it)
```
[the remark this answers is not in the document]
Comment (kevcodez) [reply studio/components/interfaces/Billing/PlanSelection/PlanSelection.utils.ts]: Adjusted

**Comment by joshenlim** [comment]
just going through this now - UI needs to change slightly on the /settings/billing/update page - with 4 cards its looking a little squeezy now, will make the changes

**Comment by joshenlim** [comment]
Decided that we'll get this through to staging first and improve the UI in subsequent PRs since this is behind a flag either way - @alaister could you help +1 this PR?

**Review by alaister — APPROVED** [review]
LGTM!

**Pull request merged**